components/job-orchestration/job_orchestration/executor/query/extract_stream_task.py

```diff
@@ -23,65 +23,153 @@
 logger = get_task_logger(__name__)
 
 
-def make_command(
+def _make_clp_command_and_env_vars(
     clp_home: Path,
     worker_config: WorkerConfig,
     archive_id: str,
     job_config: dict,
     results_cache_uri: str,
     print_stream_stats: bool,
-) -> Optional[List[str]]:
-    storage_engine = worker_config.package.storage_engine
+) -> Tuple[Optional[List[str]], Optional[Dict[str, str]]]:
+    storage_type = worker_config.archive_output.storage.type
     archives_dir = worker_config.archive_output.get_directory()
     stream_output_dir = worker_config.stream_output.get_directory()
     stream_collection_name = worker_config.stream_collection_name
 
-    if StorageEngine.CLP == storage_engine:
-        logger.info("Starting IR extraction")
-        extract_ir_config = ExtractIrJobConfig.parse_obj(job_config)
-        if not extract_ir_config.file_split_id:
-            logger.error("file_split_id not supplied")
-            return None
-        command = [
-            str(clp_home / "bin" / "clo"),
-            "i",
-            str(archives_dir / archive_id),
-            extract_ir_config.file_split_id,
+    if StorageType.S3 == storage_type:
+        logger.error(
+            f"IR extraction is not supported for storage type '{storage_type}' while using the"
+            f" '{worker_config.package.storage_engine}' storage engine."
+        )
```

Comment:
Nit: let's be consistent with the comments we used in previous PRs:

```suggestion
        logger.error(
            f"IR extraction is not supported for storage type `{storage_type}` with storage engine"
            f" `{worker_config.package.storage_engine}`."
        )
```

Reply:
This is identical to how kirk edited the related comment in the other PR actually.

Reply:
True, then let's use the one you have.